torchgeo/datamodules/reforestree.py

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""ReforesTree datamodule."""
+
+from typing import Any
+
+import kornia.augmentation as K
+import torch
+from torch.utils.data import random_split
+
+from ..datasets import ReforesTree
+from .geo import NonGeoDataModule
+from ..samplers.utils import _to_tuple
+from ..transforms.transforms import _RandomNCrop
+
+
+
+class ReforesTreeDataModule(NonGeoDataModule):
+    """LightningDataModule implementation for the ReforesTree dataset.
+
+    Implements 80/20 train/val splits.
+
+    .. versionadded:: 0.7
+        
+    """
+
+    def __init__(
+        self, 
+        batch_size: int = 64, 
+        patch_size: tuple[int, int] | int = 64, 
+        num_workers: int = 0, 
+        val_split_pct: float = 0.2,
+        test_split_pct: float = 0.2,
+        size: int = 512, 
+        **kwargs: Any
+    ) -> None:
+        """Initialize a new ReforesTreeDataModule instance.
+
+        Args:
+            batch_size: Size of each mini-batch.
+            patch_size: Size of each patch, either ``size`` or ``(height, width)``.
+                Should be a multiple of 32 for most segmentation architectures.
+            num_workers: Number of workers for parallel data loading.
+            val_split_pct: Percentage of the dataset to use as a validation set.
+            test_split_pct: Percentage of the dataset to use as a test set.
+            **kwargs: Additional keyword arguments passed to
+                :class:`~torchgeo.datasets.ReforesTree`.
+        """
+        super().__init__(ReforesTree, batch_size, num_workers, **kwargs)
```

Comment:
```suggestion
        super().__init__(ReforesTree, 1, num_workers, **kwargs)
```
A bit counter-intuitive, but each mini-batch will contain a single 4000 x 4000 px image, chopped up into $N$ smaller patches.